controller/src/test/java/io/pravega/controller/rest/v1/StreamMetaDataTests.java

```diff
@@ -743,6 +745,75 @@ public void testGetScalingEvents() throws Exception {
+                .build();
+        final StreamConfiguration readerGroup2 = StreamConfiguration.builder()
+                .scope(scope1)
+                .streamName(getStreamForReaderGroup("readerGroup2"))
+                .scalingPolicy(ScalingPolicy.fixed(1))
+                .build();
+
+        // Fetch reader groups list.
+        List<StreamConfiguration> streamsList = Arrays.asList(streamconf1, streamconf2, readerGroup1, readerGroup2);
+        when(mockControllerService.listStreamsInScope(scope1)).thenReturn(CompletableFuture.completedFuture(
+                streamsList));
+        Response response = client.target(resourceURI).request().buildGet().invoke();
+        assertEquals("List Reader Groups response code", 200, response.getStatus());
+        final ReaderGroupsList readerGroupsList1 = response.readEntity(ReaderGroupsList.class);
+        assertEquals("List count", readerGroupsList1.getReaderGroups().size(), 2);
+        assertEquals("List element", readerGroupsList1.getReaderGroups().get(0).getReaderGroupName(),
+                "readerGroup1");
+        assertEquals("List element", readerGroupsList1.getReaderGroups().get(1).getReaderGroupName(),
+                "readerGroup2");
+        response.close();
+
+        // Test for list reader groups for non-existing scope.
+        final CompletableFuture<List<StreamConfiguration>> completableFuture1 = new CompletableFuture<>();
+        completableFuture1.completeExceptionally(StoreException.create(StoreException.Type.DATA_NOT_FOUND));
+        when(mockControllerService.listStreamsInScope("scope1")).thenReturn(completableFuture1);
+        response = client.target(resourceURI).request().buildGet().invoke();
+        assertEquals("List Reader Groups response code", 404, response.getStatus());
+        response.close();
+
+        // Test for list reader groups failure.
+        final CompletableFuture<List<StreamConfiguration>> completableFuture = new CompletableFuture<>();
+        completableFuture.completeExceptionally(new Exception());
+        when(mockControllerService.listStreamsInScope("scope1")).thenReturn(completableFuture);
+        response = client.target(resourceURI).request().buildGet().invoke();
+        assertEquals("List Reader Groups response code", 500, response.getStatus());
+        response.close();
+
+        // Test to list large number of reader groups.
+        streamsList = Collections.nCopies(1000, readerGroup1);
+        when(mockControllerService.listStreamsInScope("scope1")).thenReturn(
+                CompletableFuture.completedFuture(streamsList));
+        response = client.target(resourceURI).request().buildGet().invoke();
+        assertEquals("List Reader Groups response code", 200, response.getStatus());
+        final ReaderGroupsList readerGroupsList = response.readEntity(ReaderGroupsList.class);
+        assertEquals("List count", 200, readerGroupsList.getReaderGroups().size());
+        response.close();
+    }
+
```

Comment:
Have added a system test for this, since unit test cannot cover this API entirely due to dependency on external readergroup client. Will also include this in integration tests